### DISCUSSION TITLE
Filter visualisation

### DISCUSSION THREAD

**Discussion by syedrafee** [timeseriesAI/tsai#660] [General]
Hi is there any function to visualise the filters for each layer in a 1D-CNN?

**Comment by oguiza** [comment]
No, there isn't. 
There's functionality useful to extract all layers with weights from a model:
```python
from tsai.models.utils import *
layers_w_weight = get_layers(learn.model, has_weight)
```
But then each Conv1d for example has a 3D kernel. You'll need to define a process to visualize those weights.
